File: esn_tutorial_py/esn_tutorial_py/orchestrator_node.py
import rclpy
from esn_tutorial_py.orchestrator import Orchestrator


def main(args=None):
    rclpy.init(args=args)
    node = Orchestrator()
    
    try:
        rclpy.spin(node)
    except KeyboardInterrupt:
        node.get_logger().info("User interrupted with Ctrl+C")
    finally:
        node.destroy_node()
        if rclpy.ok():
            rclpy.shutdown()



if __name__ == '__main__':
    main()

File: esn_tutorial/src/orchestrator_node.cpp
#include <esn_tutorial/orchestrator.hpp>

int main(int argc, char ** argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<esn::Orchestrator>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}
